### PR TITLE
Print inline frames for fatal stacktraces

### PR DESCRIPTION
**Pull request by abyss7**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Print inline frames for fatal stacktraces

Detailed description / Documentation draft:
Not used for exceptions because it's very slow (~10 seconds).

Sample stacktrace looks like
```
2021.01.20 17:04:23.991012 [ 7988 ] {} <Fatal> BaseDaemon: ########################################
2021.01.20 17:04:23.992093 [ 7988 ] {} <Fatal> BaseDaemon: (version 21.1.1.5646, build id: BE1BCAEDAC49815F) (from thread 7969) (query_id: f03956c1-00c1-4ced-8e5d-5cc58b2b7e81) Received signal Floating point exception (8)
2021.01.20 17:04:23.992545 [ 7988 ] {} <Fatal> BaseDaemon: Integer divide by zero.
2021.01.20 17:04:23.993063 [ 7988 ] {} <Fatal> BaseDaemon: Stack trace: 0x7f099f07dc8b 0x7f099f07cc7b 0x7f098d2a891f 0x7f098d2a9b8e 0x7f0988812bc2 0x7f0988810f41 0x7f0988f4c4ec 0x7f0988f4c44f 0x7f0988f4c40d 0x7f0988f4c3dd 0x7f0988f4c3b5 0x7f0988f4c38d 0x7f09a49ea1e9 0x7f09a49ea195 0x7f0988f4ae06 0x7f0988f4b7d9 0x7f0988f496e6 0x7f0988f489d6 0x7f0988f6fb32 0x7f0988f6fa66 0x7f0988f6f9ed 0x7f0988f6f9b1 0x7f0988f6f8f2 0x7f0988f6f7fc 0x7f0988f6f72d 0x7f0988f6f6fd 0x7f0988f6f6d5 0x7f0988f6f6b0
2021.01.20 17:04:24.302804 [ 7988 ] {} <Fatal> BaseDaemon: 4.1. inlined from gh/ch/build/../src/AggregateFunctions/AggregateFunctionSum.h:141: void DB::AggregateFunctionSumKahanData<double>::addImpl<double>(double, double&, double&)
2021.01.20 17:04:24.303073 [ 7988 ] {} <Fatal> BaseDaemon: 4. ../src/AggregateFunctions/AggregateFunctionSum.h:166: void DB::AggregateFunctionSumKahanData<double>::addMany<double>(double const*, unsigned long) @ 0x533dc8b in gh/ch/build/src/AggregateFunctions/libclickhouse_aggregate_functionsd.so
2021.01.20 17:04:24.601918 [ 7988 ] {} <Fatal> BaseDaemon: 5. gh/ch/build/../src/AggregateFunctions/AggregateFunctionSum.h:319: DB::AggregateFunctionSum<double, double, DB::AggregateFunctionSumKahanData<double>, (DB::AggregateFunctionSumType)2>::addBatchSinglePlace(unsigned long, char*, DB::IColumn const**, DB::Arena*, long) const @ 0x533cc7b in gh/ch/build/src/AggregateFunctions/libclickhouse_aggregate_functionsd.so
2021.01.20 17:04:25.072417 [ 7988 ] {} <Fatal> BaseDaemon: 6. gh/ch/build/../src/Interpreters/Aggregator.cpp:622: DB::Aggregator::executeWithoutKeyImpl(char*&, unsigned long, DB::Aggregator::AggregateFunctionInstruction*, DB::Arena*) @ 0x13dd91f in gh/ch/build/src/libclickhouse_interpretersd.so
2021.01.20 17:04:25.570065 [ 7988 ] {} <Fatal> BaseDaemon: 7. gh/ch/build/../src/Interpreters/Aggregator.cpp:765: DB::Aggregator::executeOnBlock(std::__1::vector<COW<DB::IColumn>::immutable_ptr<DB::IColumn>, std::__1::allocator<COW<DB::IColumn>::immutable_ptr<DB::IColumn> > >, unsigned long, DB::AggregatedDataVariants&, std::__1::vector<DB::IColumn const*, std::__1::allocator<DB::IColumn const*> >&, std::__1::vector<std::__1::vector<DB::IColumn const*, std::__1::allocator<DB::IColumn const*> >, std::__1::allocator<std::__1::vector<DB::IColumn const*, std::__1::allocator<DB::IColumn const*> > > >&, bool&) @ 0x13deb8e in gh/ch/build/src/libclickhouse_interpretersd.so
2021.01.20 17:04:25.857102 [ 7988 ] {} <Fatal> BaseDaemon: 8. gh/ch/build/../src/Processors/Transforms/AggregatingTransform.cpp:525: DB::AggregatingTransform::consume(DB::Chunk) @ 0x1e0bc2 in gh/ch/build/src/libclickhouse_processors_transformsd.so
2021.01.20 17:04:26.143075 [ 7988 ] {} <Fatal> BaseDaemon: 9. gh/ch/build/../src/Processors/Transforms/AggregatingTransform.cpp:495: DB::AggregatingTransform::work() @ 0x1def41 in gh/ch/build/src/libclickhouse_processors_transformsd.so
2021.01.20 17:04:26.497666 [ 7988 ] {} <Fatal> BaseDaemon: 10. gh/ch/build/../src/Processors/Executors/PipelineExecutor.cpp:79: DB::executeJob(DB::IProcessor*) @ 0xe54ec in gh/ch/build/src/libclickhouse_processors_executorsd.so
2021.01.20 17:04:26.827326 [ 7988 ] {} <Fatal> BaseDaemon: 11. gh/ch/build/../src/Processors/Executors/PipelineExecutor.cpp:96: DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0::operator()() const @ 0xe544f in gh/ch/build/src/libclickhouse_processors_executorsd.so
2021.01.20 17:04:27.159705 [ 7988 ] {} <Fatal> BaseDaemon: 12. gh/ch/build/../contrib/libcxx/include/type_traits:3676: decltype(std::__1::forward<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0&>(fp)()) std::__1::__invoke<DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0&>(DB::PipelineExecutor::addJob(DB::ExecutingGraph::Node*)::$_0&) @ 0xe540d in gh/ch/build/src/libclickhouse_processors_executorsd.so
```

### CODE REVIEW

**Comment by KochetovNicolai** [comment]
I have unassigned myself cause I am not into DWARF parsing.
This pr does pretty cool thing and code looks fine fore me except I did not fully understood what is going on there.

We probably need extra setting to show inline frames for exceptions cause it may be useful for debugging.

Also, I have a concern about this comment:
https://github.com/ClickHouse/ClickHouse/blob/06446b4f08a142d6f1bc30664c47ded88ab51782/src/Common/Dwarf.h#L48-L55

I think it is not true now, cause we use vectors in some places.


I suppose this pr may be merged if nobody else gonna review it.

**Comment by abyss7** [comment]
> I think it is not true now, cause we use vectors in some places.
> 
> I suppose this pr may be merged if nobody else gonna review it.

You are right - I missed this comment.

**Comment by abyss7** [comment]
ASAN Fuzzer: #20017

**Comment by abyss7** [comment]
Integration test results will be fixed on infrastructure side soon

**Comment by abyss7** [comment]
LGTM

File: base/daemon/BaseDaemon.cpp [hunk 1]
```diff
@@ -152,7 +152,7 @@ static void signalHandler(int sig, siginfo_t * info, void * context)
     if (sig != SIGTSTP) /// This signal is used for debugging.
     {
         /// The time that is usually enough for separate thread to print info into log.
-        sleepForSeconds(10);
+        sleepForSeconds(20);  /// FIXME: use some feedback from threads that process stacktrace
```
Comment (alexey-milovidov) [inline base/daemon/BaseDaemon.cpp:155]: Ok.

About the feedback.
There are cases when multiple threads receive signals in short period of time (example: segfaults in all query processing threads). We cannot terminate after printing only the first stack trace. Some special logic with timeouts required.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**